<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,5,2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0,1">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLineEdit" name="problom_lineEdit">
            <property name="font">
             <font>
              <family>Courier New</family>
              <pointsize>14</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="problem_pbtn">
            <property name="text">
             <string>取得題目</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="account_label">
            <property name="font">
             <font>
              <family>Aparajita</family>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>帳號</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="account_lineEdit">
            <property name="font">
             <font>
              <family>Courier New</family>
              <pointsize>14</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="password_label">
            <property name="font">
             <font>
              <family>Aparajita</family>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>密碼</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="password_lineEdit">
            <property name="font">
             <font>
              <family>Courier New</family>
              <pointsize>14</pointsize>
             </font>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="codeSubmit">
          <property name="font">
           <font>
            <family>Aparajita</family>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>上傳程式碼</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="3,1">
        <item>
         <layout class="QGridLayout" name="gridLayout"/>
        </item>
        <item>
         <widget class="QTextEdit" name="error_textEdit">
          <property name="font">
           <font>
            <family>Courier</family>
            <pointsize>14</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_1" stretch="0,0,0,0">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_1">
          <item>
           <widget class="QLabel" name="input_label">
            <property name="font">
             <font>
              <family>Courier</family>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Input</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="inputClear_pbtn">
            <property name="font">
             <font>
              <family>Courier</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="input_textEdit">
          <property name="font">
           <font>
            <family>Courier</family>
            <pointsize>14</pointsize>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="output_label">
            <property name="font">
             <font>
              <family>Courier</family>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Output</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="outputClear_pbtn">
            <property name="font">
             <font>
              <family>Courier</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextEdit" name="output_textEdit">
          <property name="font">
           <font>
            <family>Courier</family>
            <pointsize>14</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>14</pointsize>
    </font>
   </property>
   <widget class="QMenu" name="menu_F">
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>檔案(&amp;F)</string>
    </property>
    <addaction name="actionNew_Project"/>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>編輯(&amp;E)</string>
    </property>
    <addaction name="actionCopy"/>
    <addaction name="actionCut"/>
    <addaction name="actionPaste"/>
    <addaction name="separator"/>
    <addaction name="actionSelect_All"/>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
   </widget>
   <widget class="QMenu" name="menu_C">
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>編譯(&amp;C)</string>
    </property>
    <addaction name="actionBuild"/>
    <addaction name="actionRun"/>
    <addaction name="actionBuild_and_run"/>
   </widget>
   <widget class="QMenu" name="menu_S">
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="title">
     <string>排版(&amp;S)</string>
    </property>
    <addaction name="actionAStyle"/>
   </widget>
   <addaction name="menu_F"/>
   <addaction name="menu_E"/>
   <addaction name="menu_C"/>
   <addaction name="menu_S"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="font">
    <font>
     <family>Agency FB</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew_Project"/>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="separator"/>
   <addaction name="actionSave"/>
   <addaction name="actionSave_As"/>
   <addaction name="separator"/>
   <addaction name="actionCopy"/>
   <addaction name="actionCut"/>
   <addaction name="actionPaste"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionBuild"/>
   <addaction name="actionRun"/>
   <addaction name="actionBuild_and_run"/>
   <addaction name="separator"/>
   <addaction name="actionAStyle"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/filenew.png</normaloff>:/images/images/filenew.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/fileopen.png</normaloff>:/images/images/fileopen.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/filesave.png</normaloff>:/images/images/filesave.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/filesaveas.png</normaloff>:/images/images/filesaveas.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/editcopy.png</normaloff>:/images/images/editcopy.png</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/editcut.png</normaloff>:/images/images/editcut.png</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/editpaste.png</normaloff>:/images/images/editpaste.png</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionSelect_All">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/undo.png</normaloff>:/images/images/undo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/redo.png</normaloff>:/images/images/redo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionBuild">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/compile.png</normaloff>:/images/images/compile.png</iconset>
   </property>
   <property name="text">
    <string>Build</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/run.png</normaloff>:/images/images/run.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionBuild_and_run">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/compilerun.png</normaloff>:/images/images/compilerun.png</iconset>
   </property>
   <property name="text">
    <string>Build and run</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionAStyle">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/format.png</normaloff>:/images/images/format.png</iconset>
   </property>
   <property name="text">
    <string>AStyle</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="actionNew_Project">
   <property name="icon">
    <iconset resource="AppResources.qrc">
     <normaloff>:/images/images/newproject.png</normaloff>:/images/images/newproject.png</iconset>
   </property>
   <property name="text">
    <string>New Project</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="AppResources.qrc"/>
 </resources>
 <connections/>
</ui>
